### ISSUE TITLE
bootstrap could automatically enable --use-ssh-agent when SSH_AUTH_SOCK is present

### ISSUE THREAD

**Issue by zzhou1** [ClusterLabs/crmsh#1633]
Here is the current oneliner:

adm@tw-1:~> sudo -E crm cluster init --use-ssh-agent -ys /dev/disk/by-partlabel/sbd -N $USER@tw-2

It would be naturally consistent with "ssh", "ssh-copy-id", etc. That means as long as SSH_AUTH_SOCK gets detected, --use-ssh-agent is automatically enabled. We can add code to enforce `--use-ssh-agent no` if really need to use the local ssh key.

adm@tw-1:~> sudo -E crm cluster init -ys /dev/disk/by-partlabel/sbd -N $USER@tw-2

**Comment by liangxin1300** [comment]
Don't get your point yet.
> adm@tw-1:~> sudo -E crm cluster init --use-ssh-agent -ys /dev/disk/by-partlabel/sbd -N $USER@tw-2

Use `--use-ssh-agent` to tell crmsh we want to enable ssh-agent
If we don't use `--use-ssh-agent` option, local ssh key will be created and used:
```
adm@alp-1:~> sudo -E crm cluster init -ys /dev/sda9 -N adm@alp-2
INFO: Loading "default" profile from /etc/crm/profiles.yml
INFO: Loading "knet-default" profile from /etc/crm/profiles.yml
WARNING: $SSH_AUTH_SOCK is detected. As a tip, using the --use-ssh-agent option could avoid generate local root ssh keys on cluster nodes.
INFO: A new ssh keypair is generated for user adm.
INFO: A new ssh keypair is generated for user hacluster.
```

**Comment by nicholasyang2022** [comment]
> Auto enable ssh agent when SSH_AUTH_SOCK is present

Yeah, I am going to do this, as it is consistent with the behavior of `ssh-copy-id`.

**Comment by nicholasyang2022** [comment]
These changes are needed:

* Default to try to use ssh-agent.
* When ssh-agent is not available, do not fail and fallback to use or generate local key files.
* When there are no keys in the ssh-agent, do not fail and fallback to use or generate local key files.
